RadixWalletTests/Clients/ProfileStoreTests/ProfileStoreTests.swift

```diff
@@ -1,277 +1,1122 @@
+import DependenciesAdditions
 @testable import Radix_Wallet_Dev
 import XCTest
 
-// MARK: - ProfileStoreTests
-final class ProfileStoreTests: TestCase {
-	/// This test method has not been implemented with any particular thoughts in mind
-	/// mostly creating a bunch of unstructured task and reading/setting profile on shared
-	/// profile store interleaved with some `Task.yield()`'s which resulted in failures before
-	/// ProfileStore was migrated to use ManagedAtomicLazyReference, for more reading about this
-	/// see: https://forums.swift.org/t/is-this-an-ok-solution-to-achieve-shared-instance-of-actor-using-async-init/63528/2
-	func test_assert_ProfileStore_is_reentrance_free() async throws {
-		try await withDependencies {
-			#if canImport(UIKit)
-			$0.device.$name = deviceName
-			$0.device.$model = deviceModel.rawValue
-			#endif
-			$0.uuid = .incrementing
-			$0.mnemonicClient.generate = {
-				XCTAssertNoDifference($0, BIP39.WordCount.twentyFour)
-				XCTAssertNoDifference($1, BIP39.Language.english)
-				return .testValue
-			}
-			$0.secureStorageClient.saveMnemonicForFactorSource = { XCTAssertNoDifference($0.factorSource.kind, .device) }
-			$0.secureStorageClient.loadProfileSnapshotData = { _ in nil }
-			$0.secureStorageClient.loadDeviceIdentifier = {
-				.init(uuidString: "BABE1442-3C98-41FF-AFB0-D0F5829B020D")!
-			}
-			$0.secureStorageClient.getDeviceIdentifierSetIfNil = {
-				$0
-			}
-			$0.date = .constant(Date(timeIntervalSince1970: 0))
-			$0.userDefaultsClient.stringForKey = { _ in
-				"BABE1442-3C98-41FF-AFB0-D0F5829B020D"
+// swiftformat:disable redundantInit
+
+extension DependencyValues {
+	private mutating func _profile(_ profile: Profile?) {
+		secureStorageClient.loadProfile = { _ in
```

Comment:
that would be testing two layers simultaneously, and is a testing anti-pattern. We should only test one layer at a time.

Reply:
I would worry more about the profileStore not working properly in integration with the Keychain, rather than worry about going against an anti-pattern. 

IMO, testing the two separately, ProfileStore and Keychain, does not guarantee that they would work properly when put together. Do we plan to write have some integration tests on another occasion, or we would rely on testing with the real App?

Reply:
think perhaps we could use Reducer tests for that purpose (what you call "integration test") and use **live** AccountsClient/FactorSourcesClient etc which will be using `ProfileStore` and then as you say, use a test Keychain client.

Reply:
Sure, but if we cannot test ProfileStore with a test keychain easily, the same will happen for those clients.

Reply:
we can, it is just a dependency... not sure what your point is?

Reply:
I mean testing with a real Keychain, not with mock of the KeychainClient.

Currently, we only test with a real Keychain when running the App. Given how crucial is this part of the app, it would definitely benefit to have tests against a real Keychain, if this is possible.

When ProfileStore mutates a profile, it really would be best to test that it is actually saved in the Keychain, by loading it back and asserting against it, instead of mocking the secureStorageClient save and load profile functions.

Reply:
iOS Simulator Keychain is not a real keychain, so cannot be done.

Reply:
But in the cake with 6 layers:
CreateAccount+Reducer
AccountsClient
ProfileStore
SecureStorageClient
KeychainClient
Keychain

We can test all of the top 5 layers! That is very good! and should be enough. If our KeychainClient+Live impl is thin and stupid enough (it is!) then we are good, since we use KeychainAccess